rfcs/0048-on-conflict-clause-in-create-table-statement.md

```diff
@@ -0,0 +1,121 @@
+---
+feature: ON CONFLICT Clause in Create Table Statement
+authors:
+  - "st1page"
+start_date: "2023/2/1"
+---
+
+# ON CONFLICT Clause in Create Table Statement
+
+## Summary
+
+Allow user to declare the conflict behavior when the newly inserted row break the unique constraint of primary key.
+
+## Motivation
+
+Currently, for a table with primary key, the later operations will always overwrite the previous ones when the new operation [0017-DML.md](0017-DML.md)
+But users' requirement are various here.
+I will give some different behavior here. Notice that here the conflict behavior is not just happens when an insert DML statement happening but also in a table with external connector.
+
+- **Overwrite**
+  The most important needs of using the Overwrite behavior is to express "Upsert" operation. Some connector can not give the old value of the update operation and we need to overwrite the old value with the same pk in storage.
+
+- **Ignore**
+  The Ignore behavior make user can define a "append-only table with primary key" which maintain the primary constraint and also provide a append-only changes to downstream with good performance. This is good to express a table with an At-least-once Delivery connector where a same record can be replayed multiple times to break the primary key constraint.
```

Comment:
Are we talking about `CREATE TABLE`? As we all agreed, tables must be updatable, so "append-only changes" here do not make senses. (because we must allow user to revise data)

Reply:
```SQL
create table t(v int, k int) APPEND ONLY; -- we will not expose this to user

create source s(v int, k int);
create materialized view mv as select * distinct on(k) from s;
```